### PR TITLE
`assets.order` now initializes the agreement and makes the payment in just one contract call

### PR DESCRIPTION
**Pull request by aaitor**
## Description

see title

## Is this PR related with an open issue?

Related to Issue #220

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
 
## Checklist:

- [x] Follows the code style of this project.
- [x] Tests Cover Changes
- [ ] Documentation

### CODE REVIEW

File: src/nevermined/Assets.ts [hunk 1]
```diff
@@ -977,69 +980,34 @@ export class Assets extends Instantiable {
...
-                    this.logger.error('Agreement ID: ', agreementId)
-                    this.logger.error('DID: ', ddo.id)
-                    reject(new Error('Error on payment'))
-                }
+            const template = keeper.getAccessTemplateByName(templateName)
+            const assetRewards = getAssetRewardsFromService(service)
 
-                resolve(did)
-            })
+            console.log(`Template: ${JSON.stringify(templateName)}`)
```
Comment (r-marques) [inline src/nevermined/Assets.ts:993]: maybe remove this or use the logger

Comment (aaitor) [reply src/nevermined/Assets.ts:993]: forgot to remove that one, good catch

**Comment by mrsmkl** [comment]
Looking good, but there's probably a conflict with my last PR that was merged...

**Review by r-marques — APPROVED** [review]
lgtm

**Comment by aaitor** [comment]
> Looking good, but there's probably a conflict with my last PR that was merged...

Ya, moving the initialization of the KeyTransfer to the method broke the compilation, but it should be already fixed

**Pull request merged**